cpp/tests/io/parquet_test.cpp

```diff
@@ -3348,7 +3348,7 @@ TEST_F(ParquetWriterTest, CheckPageRows)
   // check first page header and make sure it has only 5000 values
   int fd = open(filepath.c_str(), O_RDONLY);
   unsigned char buf[1024];
-  read(fd, buf, sizeof(buf));
+  (void)!read(fd, buf, sizeof(buf));
```

Comment:
```suggestion
  size_t read_bytes = read(fd, buf, sizeof(buf));
  EXPECT_EQ(read_bytes, sizeof(buf));
```

Reply:
Wait, I'm not very familiar with `read`. Does that statement read exactly `1024` bytes? What if the file is smaller?

Reply:
No, `read_bytes` is the actual size of read file, while `sizeof(buff)` is `1024`. Do we know the file size before hand?

Reply:
> It should read 1024 bytes unless the file is shorter or there is an error.

Do we know the file size is >1024?

Reply:
Ah I see, file size should be bigger. But this is error-prone since we may change the file size generated above at any time.

Reply:
I didn't read into what is done with this buffer, but if we expect the buffer to contain 1024 bytes of real data, the size read needs to match the buffer size here. If we only look at the first x bytes of the buffer, why isn't it that size?

Reply:
That `read` command tries to read to the full buffer size, but it may stop if the file size is smaller than the buffer, and return the number of actual bytes read.

Reply:
Yes, exactly. If the file size is shorter, why do we try to read this many bytes? If we expect 1k bytes to exist in the file, we can read 1k bytes like this. We still need to be careful to not interpret random memory as read bytes though. File reads can and do fail on occasion. People eject the Bluray on the PS3 at crazy times.

By checking that our buffer is fully read we verify that the data we are about to look at is valid and correct. If we only care about a header on the file and not 1k bytes, we should size this buffer appropriately and read that many bytes, unless this is a performance issue for the hardware. Without specific targeting of hardware, this isn't possible to know.

I believe that reading the buffer full and verifying the full buffer is there is the correct path here.

Reply:
This test is confirming that the first data page in the written parquet file is actually limited to the number of rows set in the args (5000). It seems that it is sufficient to read the first few bytes of the file to check the valid header, and the page metadata. Since this information rarely crosses the 1024 byte mark in the file, this test seems to work. 

There is provision to skip the first page if it is not a data page (dictionary pages wouldn't be affected by the `max_page_size_rows()` argument). In that case, there is an additional read of 1024 starting from where the second page begins. 

I agree that this logic could be made cleaner with c++ streams but for now, I don't see the test breaking on its own.
@etseidl (author) can confirm.

Reply:
Sorry I can't reproduce any compiler warning/failure with gcc (https://godbolt.org/z/3EY9Kx7aK).
Can anybody give any idea?

Reply:
I can't seem to reproduce the warning either even with `-Wall` and `-std=c++17`.

Reply:
> I agree that this logic could be made cleaner with c++ streams but for now, I don't see the test breaking on its own. Unless, of course, if the read fails altogether for some reason. @etseidl (author) can confirm.

Sorry, coming late to the game...I have changes to this test pending in a separate branch.  PR will be coming as soon as I get work approval.  I've completely reworked this test to use the cudf::io::datasource interface with the CompactProtocolReader.  

Sorry for the trouble!

Reply:
If it is coming in 22.08, I wouldn't bother to pull it out.

Reply:
I am using `c++ (Ubuntu 9.4.0-1ubuntu1~18.04) 9.4.0` in my local machine

I cannot replicate in godbolt. But can replicate same error in my local machine compilation
`g++ -fdiagnostics-color=always -Wno-deprecated-declarations -O3 -DNDEBUG -fPIE -Wall -Werror -Wno-unknown-pragmas -Wno-error=deprecated-declarations -Wno-deprecated-declarations -pthread -std=gnu++17 test.cpp`
```cpp
#include <string>
#include <unistd.h>
#include <fcntl.h>

int main()  {
     std::string filepath = "CheckPageRows.parquet";
     int fd = open(filepath.c_str(), O_RDONLY);
     unsigned char buf[1024];
     read(fd, buf, sizeof(buf));
     return 0;
}
```

```
test.cpp: In function ‘int main()’:
test.cpp:9:10: error: ignoring return value of ‘ssize_t read(int, void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
    9 |      read(fd, buf, sizeof(buf));
      |      ~~~~^~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

Reply:
```bash
(base) yunsongw@yunsongw-dt:~/Work$ gcc --version
gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

(base) yunsongw@yunsongw-dt:~/Work$ g++ -Werror -O toto.cxx 
toto.cxx: In function ‘int main()’:
toto.cxx:9:10: error: ignoring return value of ‘ssize_t read(int, void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
    9 |      read(fd, buf, sizeof(buf));
      |      ~~~~^~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```
`-Werror -O` is the minimum to replicate the error on my local system but I cannot reproduce the same error with any of the godbolt compiler versions (https://godbolt.org/z/f5aTzM7fx).